### PR TITLE
Various code logic corrections.

### PR DESCRIPTION
**Pull request by Hybris95**
### Description of Change ###

Corrections after passing a PVS-Studio on the Project
No tests because no test environment yet just newly forking the project.

Some corrections may be implemented later (View.cs & LayoutAlignmentExtensions.cs).
Tests needs to be done also.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/Thickness.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public double VerticalThickness
 
 		internal bool IsDefault
 		{
-			get { return Left == 0 && Top == 0 && Right == 0 && Left == 0; }
+			get { return Left == 0 && Top == 0 && Right == 0 && Bottom == 0; }
```
Comment (ghuntley) [inline Xamarin.Forms.Core/Thickness.cs]: :+1:

Comment (jassmith) [inline Xamarin.Forms.Core/Thickness.cs]: woops

File: Xamarin.Forms.Core/ItemsView.cs [hunk 2]
```diff
@@ -1,8 +1,9 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 
 namespace Xamarin.Forms
 {
-	public abstract class ItemsView<TVisual> : View, IItemsView<TVisual> where TVisual : BindableObject
+	public abstract class ItemsView<TVisual> : IDisposable, View, IItemsView<TVisual> where TVisual : BindableObject
```
Comment (jassmith) [inline Xamarin.Forms.Core/ItemsView.cs]: This doesn't compile :-1:

File: Xamarin.Forms.Core/MultiPage.cs [hunk 3]
```diff
@@ -8,7 +8,7 @@
 namespace Xamarin.Forms
 {
 	[ContentProperty("Children")]
-	public abstract class MultiPage<T> : Page, IViewContainer<T>, IPageContainer<T>, IItemsView<T> where T : Page
+	public abstract class MultiPage<T> : Page, IDisposable, IViewContainer<T>, IPageContainer<T>, IItemsView<T> where T : Page
```
Comment (jassmith) [inline Xamarin.Forms.Core/MultiPage.cs]: Same as above

**Comment by jassmith** [comment]
This PR adds API (specifically it marks things as disposable). Please re-submit without API changes and build fixes.

**Comment by Hybris95** [comment]
Ill check this out and resubmit.

**Pull request closed without merging**